examples/10_HasegawaWakatani/10_hasegawa_wakatani.cpp

```diff
@@ -0,0 +1,734 @@
+// SPDX-FileCopyrightText: (C) The Kokkos-FFT development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0 WITH LLVM-exception
+
+#include <memory>
+#include <random>
+#include <filesystem>
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Complex.hpp>
+#include <Kokkos_Random.hpp>
+#include <KokkosFFT.hpp>
+#include "io_utils.hpp"
+
+#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP) || \
+    defined(KOKKOS_ENABLE_SYCL)
+constexpr int TILE0 = 4, TILE1 = 32;
+#else
+constexpr int TILE0 = 4, TILE1 = 4;
+#endif
+
+using execution_space = Kokkos::DefaultExecutionSpace;
+template <typename T>
+using View1D = Kokkos::View<T*, Kokkos::LayoutRight, execution_space>;
+template <typename T>
+using View2D = Kokkos::View<T**, Kokkos::LayoutRight, execution_space>;
+template <typename T>
+using View3D = Kokkos::View<T***, Kokkos::LayoutRight, execution_space>;
+
+// \brief A class to represent the grid used in the Hasegawa-Wakatani model.
+struct Grid {
+  //! Grid in x direction (nkx * 2 + 1)
+  View1D<double> m_kx;
+
+  //! Grid in y direction (nky + 1)
+  View1D<double> m_kyh;
+
+  //! Grid in x and y direction (nky + 1, nkx * 2 + 1)
+  View2D<double> m_ksq;
+
+  // \brief Constructor of a Grid class
+  // \param nx [in] Number of grid points in the x-direction.
+  // \param ny [in] Number of grid points in the y-direction.
+  // \param lx [in] Length of the domain in the x-direction.
+  // \param ly [in] Length of the domain in the y-direction.
+  Grid(int nx, int ny, double lx, double ly) {
+    int nkx = (nx - 2) / 3, nky = (ny - 2) / 3;
+    int nkx2 = nkx * 2 + 1, nky2 = nky * 2 + 1, nkyh = nky + 1;
+    m_kx  = View1D<double>("m_kx", nkx2);
+    m_kyh = View1D<double>("m_kyh", nkyh);
+    m_ksq = View2D<double>("m_ksq", nkyh, nkx2);
+
+    using host_execution_space = Kokkos::DefaultHostExecutionSpace;
+    // [0, dkx, 2*dkx, ..., nkx * dkx, -nkx * dkx, ..., -dkx]
+    double dkx = lx / static_cast<double>(nkx2);
+    auto h_kx  = KokkosFFT::fftfreq(host_execution_space(), nkx2, dkx);
+
+    // [0, dky, 2*dky, ..., nky * dky]
+    double dky = ly / static_cast<double>(nky2);
+    auto h_kyh = KokkosFFT::rfftfreq(host_execution_space(), nky2, dky);
+
+    // kx**2 + ky**2
+    auto h_ksq = Kokkos::create_mirror_view(m_ksq);
+    for (int iky = 0; iky < nkyh; iky++) {
+      for (int ikx = 0; ikx < nkx2; ikx++) {
+        h_ksq(iky, ikx) = h_kx(ikx) * h_kx(ikx) + h_kyh(iky) * h_kyh(iky);
+      }
+    }
```

Comment:
What about using a MDRangePolicy on the `DefaultExecutionSpace` if we use `KokkosFFT::fftfreq(execution_space(), nkx2, dkx)` ?

Reply:
If we call the GPU kernels in a constructor, it does not compile on NVIDIA GPUs. So, I would like to initialize on host and copy back to device.

Reply:
I see, I guess the proper way would be create free functions to do this work.

Let's keep it like that for now.